### ISSUE TITLE
using a factor variable as frame aesthetic

### ISSUE THREAD

**Issue by juliasilge** [thomasp85/gganimate#4]
I am so happy about this package! So great. I am having no problems when using an integer (such as year) for the frame aesthetic, but when I try to use any factor variable for the frame aesthetic, it is using the underlying integer instead of the factor values and not finding any data. For example, if I try the code from the README to animate over continents,

```
p <- ggplot(gapminder, aes(gdpPercap, lifeExp, size = pop, frame = continent)) +
  geom_point() +
  scale_x_log10()

gg_animate(p)
```

I cannot reproduce the plot. It makes the plot with the correct axes and legend, but instead of the correct word title, it puts the integer that is underneath the factor level. Also, no data is plotted, I guess because it is trying to match the integer instead of the factor level? I've attached one of the images it makes.

![rplot](https://cloud.githubusercontent.com/assets/12505835/12859588/5ae93910-cc14-11e5-9d2f-095e0381e0ed.png)

Here is my session info, in case it is something on my end:

```
R version 3.2.3 (2015-12-10)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.11.2 (El Capitan)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gapminder_0.2.0 gganimate_0.1   ggplot2_2.0.0  

loaded via a namespace (and not attached):
 [1] labeling_0.3     colorspace_1.2-6 scales_0.3.0     plyr_1.8.3       tools_3.2.3      gtable_0.1.2    
 [7] Rcpp_0.12.3      grid_3.2.3       knitr_1.12.3     digest_0.6.9     munsell_0.4.2 
```

**Comment by timelyportfolio** [comment]
@juliasilge I believe you are asking for this to be directly handled by `gganimate`, but you can accomplish this by explicitly using `as.character(continent)`.

```
p <- ggplot(gapminder, aes(gdpPercap, lifeExp, size = pop, frame = as.character(continent))) +
  geom_point() +
  scale_x_log10()

gg_animate(p)
```

**Comment by dgrtwo** [comment]
@juliasilge is right, this should work! While changing to as.character would make the title work, it also prevents ordering based on an ordered factor.

I realized this is because of my fix in #2! I will find an approach that works with both. Thanks for the report!

**Comment by ptoche** [comment]
I have something trivial to say, but since this thread comes up in the google I thought it might be useful to comment here. Converting a factor to a character before passing it to frame works, but the order is obviously not the same as converting to numeric. In my use case, ``frame = "10"`` immediately followed ``frame = "1"`` instead of coming between 9 and 11. The problem was pretty obvious. Converting to numeric, ``as.numeric(as.character())`` produced the frames in the order expected. 

It also gave me warning messages about ``NAs introduced by coercion``, but those were "ignorable".
